app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotService.java

```diff
@@ -0,0 +1,93 @@
+package uk.ac.ebi.atlas.experimentpage.cellplot;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.math.util.MathUtils;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.experimentpage.metadata.CellMetadataDao;
+import uk.ac.ebi.atlas.experimentpage.tsne.TSnePoint;
+
+import java.util.Map;
+import java.util.LinkedHashMap;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+
+@Component
```

Comment:
I know it's minor. But it's a good time to rename as we changed the service name.